src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -230,9 +240,21 @@ object Directory {
     // Hashmap from directory path to Directory object, used below for quick access from path.
     val pathToDirectory = HashMap[Path, Directory]()
 
+    // add size hint for performance improvement.
+    if(fileIdTracker.isInstanceOf) {
+      fileIdTracker.get.addSizeHint(files.length)
+    }
+
     for ((dirPath, files) <- leafDirToChildrenFiles) {
-      val allFiles = ListBuffer[FileInfo]()
-      allFiles.appendAll(files.map(FileInfo(_)))
+      val allFiles = fileIdTracker match {
+        case Some(tracker) =>
+          files.map {
+            f =>
+              val id = tracker.addFile(f)
+              FileInfo(f, id)
```

Comment:
```scala
val allFiles = fileIdTracker.map { tracker =>
  files.map(f => FileInfo(f, tracker.addFile(f)))
}.getOrElse(files.map(FileInfo(_)))
```
seems easy enough to understand?

Reply:
Can you also update the command for `@param fileIdTracker` that it will be updated? (Btw, is that the intention? rarely see this pattern in Scala unless function name is very explicit about it).

Reply:
Updated the comment. We generate the file-ids as we list the source files here. The advantage is that id assignment happens here in once place and now fileIdTracker is the wrapper around the whole generation and Map.